nav2_dynamic_params/src/dynamic_params_validator.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+#include <limits>
+#include "nav2_dynamic_params/dynamic_params_validator.hpp"
+
+namespace nav2_dynamic_params
+{
+
+DynamicParamsValidator::DynamicParamsValidator(rclcpp::Node::SharedPtr node)
+: node_(node)
+{
+  node_->register_param_change_callback(
+    std::bind(&DynamicParamsValidator::param_validation_callback, this, std::placeholders::_1));
+}
+
+rcl_interfaces::msg::SetParametersResult DynamicParamsValidator::param_validation_callback(
+  std::vector<rclcpp::Parameter> parameters)
+{
+  auto result = rcl_interfaces::msg::SetParametersResult();
+  result.successful = true;
+
+  for (auto parameter : parameters) {
+    RCLCPP_INFO(node_->get_logger(),
+      "Parameter Change Request: %s", (parameter.get_name()).c_str());
+
+    // Validate Parameter Type
+    for (std::map<std::string, rclcpp::ParameterType>::iterator it = param_map_.begin();
+      it != param_map_.end(); ++it)
+    {
+      if (validate_param(it->first, parameter, it->second)) {
+        result.successful &= true;
+      } else {
+        result.successful = false;
+      }
+    }
+
+    // Validate Parameter Bounds
+    for (std::map<std::string, std::pair<double, double>>::iterator it = param_bound_map_.begin();
```

Comment:
Use range based for here and above
```suggestion
    for (const auto& i :  param_bound_map_)
```